crates/holochain_trace/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+//     if std::env::var_os("RUST_LOG").is_none() {
+//         return Ok(());
+//     }
+//     init_fmt(Output::OpenTel)
+// }
+
+/// Same as test_run but with timed spans
+pub fn test_run_timed() -> Result<(), errors::TracingError> {
+    if std::env::var_os("RUST_LOG").is_none() {
+        return Ok(());
+    }
+    init_fmt(Output::LogTimed)
+}
+
+/// Same as test_run_timed but saves as json
+pub fn test_run_timed_json() -> Result<(), errors::TracingError> {
+    if std::env::var_os("RUST_LOG").is_none() {
+        return Ok(());
+    }
+    init_fmt(Output::JsonTimed)
+}
+
+/// Generate a flamegraph from timed spans "busy time".
+/// Takes a path where you are piping the output into.
+/// If the path is provided a flamegraph will automatically be generated.
+/// TODO: Get auto inferno to work
+/// for now use (fish, or the bash equiv):
+/// `2>| inferno-flamegraph > flamegraph_test_ice_(date +'%d-%m-%y-%X').svg`
+/// And run with `cargo test --quiet`
+pub fn test_run_timed_flame(path: Option<&str>) -> Result<Option<impl Drop>, errors::TracingError> {
+    if std::env::var_os("RUST_LOG").is_none() {
+        return Ok(None);
+    }
+    init_fmt(Output::FlameTimed)?;
+    Ok(path.and_then(|p| {
+        toml_path().map(|mut t| {
+            t.push(p);
+            FlameTimed::new(t)
+        })
+    }))
+}
+
+/// Generate a flamegraph from timed spans "idle time".
+/// Takes a path where you are piping the output into.
+/// If the path is provided a flamegraph will automatically be generated.
+/// TODO: Get auto inferno to work
+/// for now use (fish, or the bash equiv):
+/// `2>| inferno-flamegraph -c blue > flamegraph_test_ice_(date +'%d-%m-%y-%X').svg`
+/// And run with `cargo test --quiet`
+pub fn test_run_timed_ice(path: Option<&str>) -> Result<Option<impl Drop>, errors::TracingError> {
+    if std::env::var_os("RUST_LOG").is_none() {
+        return Ok(None);
+    }
+    init_fmt(Output::IceTimed)?;
+    Ok(path.and_then(|p| {
+        toml_path().map(|mut t| {
+            t.push(p);
+            FlameTimed::new(t)
+        })
+    }))
+}
+
+/// This checks RUST_LOG for a filter but doesn't complain if there is none or it doesn't parse.
+/// It then checks for CUSTOM_FILTER which if set will output an error if it doesn't parse.
+pub fn init_fmt(output: Output) -> Result<(), errors::TracingError> {
+    let mut filter = match std::env::var("RUST_LOG") {
+        Ok(_) => EnvFilter::from_default_env(),
+        Err(_) => EnvFilter::from_default_env().add_directive("[wasm_debug]=debug".parse()?),
+    };
+    if std::env::var("CUSTOM_FILTER").is_ok() {
+        EnvFilter::try_from_env("CUSTOM_FILTER")
+            .map_err(|e| eprintln!("Failed to parse CUSTOM_FILTER {:?}", e))
+            .map(|f| {
+                filter = f;
+            })
+            .ok();
+    }
+
+    let subscriber = FmtSubscriber::builder()
+        .with_writer(std::io::stderr)
+        .with_file(true)
+        .with_line_number(true)
```

Comment:
other than disabling the opentelemetry option and updating deps, these `with_file` and `with_line_number` are the only changes I made while pulling in the observability crate

Reply:
```
[2023-03-14T13:52:54.504231459-06:00] INFO - [HOLOCHAIN Custom Binary] 2023-03-14T19:52:54.504022Z DEBUG tx5::state: /home/neonphog/projects/holochain/.cargo/registry/src/github.com-1ecc6299db9ec823/tx5-0.0.1-alpha.5/src/state.rs:435: begin register with signal server state_uniq=xdt86x9L:1 sig_url=wss://holotest.net/
```

Unfortunately, the filenames come out rather large when they are from a crates.io dependency, but I still think it's worth it to easily debug problems.